client/tso_client.go

```diff
@@ -296,6 +350,28 @@ func (c *tsoBaseClient) switchPrimary(addrs []string) error {
 	for _, cb := range c.primarySwitchedCallbacks {
 		cb()
 	}
-	log.Info("[tso] switch primary", zap.String("new-primary", addr), zap.String("old-primary", oldPrimary))
+	log.Info("[pd(tso)] switch primary", zap.String("new-primary", addr), zap.String("old-primary", oldPrimary))
 	return nil
 }
+
+func (c *tsoMcsClient) updateMember() error {
+	resp, err := c.metacli.Get(c.ctx, []byte(c.primaryKey))
+	if err != nil {
+		log.Error("[pd(tso)] failed to get the keyspace serving endpoint", errs.ZapError(err))
+		return err
+	}
+
+	if resp == nil || len(resp.Kvs) == 0 {
+		log.Error("[pd(tso)] didn't find the keyspace serving endpoint")
+		return errs.ErrClientGetLeader
+	} else if resp.Count > 1 {
+		return errs.ErrClientGetMultiResponse.FastGenByArgs(resp.Kvs)
+	}
+
+	value := resp.Kvs[0].Value
+	member := &pdpb.Member{}
+	if err := proto.Unmarshal(value, member); err != nil {
+		return errs.ErrClientProtoUnmarshal.Wrap(err).GenWithStackByCause()
+	}
+	return c.switchPrimary(addrsToUrls([]string{member.Name}))
```

Comment:
member.Name?

Reply:
That's a hack previously by reusing pdpb.Member. My next pr will introduce a new struct as we discussed before.